### PR TITLE
First pass at sorting gha dashboard

### PR DESCRIPTION
**Pull request by damccorm**
This attempts to split our dashboard into categories. Obviously its quite large, so I'll explain the methodology used to generate this and we can go from there with a heavier code review on the more potentially error prone spots.

I'll also note that I'm planning on making this better over time, so the splitting of things into labels should end up being more precise than it is in this iteration. 

How I did this:
1) All changes to sync_workflows were manual and should be reviewed normally. The goal here is to add a `dashboard_category` which maps to one of `core_<lang>`, `<runner>_<lang>`, or `misc` so that we can split tests into these categories.
2) To generate the dashboard changes (the bulk of the changes), I used the UI:

- First, I copied the existing dashboard 12 times using grafana's `duplicate` functionality.
- Next, I modified the title attribute in the UI for each of the dashboards. These are reflected in `"title"` attributes in the json (may be easier to find all instances by CMD+F searching for `"transformations"` and inspecting the previous line.
- Last, I modified each table's query (CMD+F `"rawSql"`) to add a `where` clause. All of these were of the form:

```
  where
    dashboard_category = 'core_java'
```

The split up dashboards at least render in my local grafana instance, though its hard to verify that they're doing exactly what they're supposed to do.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by damccorm** [comment]
R: @Abacn

**Review by Abacn — COMMENTED** [review]
Thanks for the work! I'm able to load the dashboard and have a preview (without data)

File: .test-infra/metrics/sync/github/sync_workflows.py [hunk 1]
```diff
@@ -55,6 +55,26 @@ def __init__(self, id, name, filename):
     self.filename = filename
     self.runs = []
 
+def get_dashboard_category(workflow_name):
```
[the remark this answers is not in the document]
Comment (damccorm) [reply .test-infra/metrics/sync/github/sync_workflows.py:58]: ;tldr - I agree we'll probably need improvements, though I'm not sure which ones make sense yet. I'd like to see how it looks now to make a smarter decision, so my vote would be to merge and iterate.

> this is generally good though for cross-lang it is subtle, e.g. "PythonUsingJava". We can always come back latter.

Agreed - I thought about adding that and decided to leave it for now. I think we'll have a much clearer picture of what further breakdowns are needed once we have this view.

> What's your opinion of keep current structure (used in workflow MDs for GHA or Jenkins) that is having PreCommit PostCommit LoadTest PerformanceTest Others

I'm interested in doing something like this, but not sure we should yet; I tend to think people looking at this dashboard will be one of the following:

1) Someone looking for an individual test (I think this meets that need)
2) Someone looking for systematic patterns in an area.

For (2), I think this meets the need, but I'm not sure. Once we have the split (and have verified it's reasonable) I'd like to gather feedback from the community

**Review by Abacn — APPROVED** [review]

File: .test-infra/metrics/sync/github/sync_workflows.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Abacn) [reply .test-infra/metrics/sync/github/sync_workflows.py:58]: yeah get your point. I only considered what maintainers need i.e. monitoring. For developers of specific components they may better guided by topic

**Pull request merged**